packages/mdl-snackbar/foundation.js

```diff
@@ -0,0 +1,142 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {MDLFoundation} from 'mdl-base';
+import {cssClasses, strings} from './constants';
+
+export default class MDLSnackbarFoundation extends MDLFoundation {
+  static get cssClasses() {
+    return cssClasses;
+  }
+
+  static get strings() {
+    return strings;
+  }
+
+  static get defaultAdapter() {
+    return {
+      addClass: (/* className: string */) => {},
+      removeClass: (/* className: string */) => {},
+      hide: () => {},
+      unhide: () => {},
+      setMessageText: (/* message: string */) => {},
+      setActionText: (/* actionText: string */) => {},
+      hideAction: () => {},
+      unhideAction: () => {},
+      registerActionClickHandler: (/* handler: EventListener */) => {},
+      deregisterActionClickHandler: (/* handler: EventListener */) => {},
+      registerTransitionEndHandler: (/* handler: EventListener */) => {},
+      deregisterTransitionEndHandler: (/* handler: EventListener */) => {}
+    };
+  }
+
+  constructor(adapter) {
+    super(Object.assign(MDLSnackbarFoundation.defaultAdapter, adapter));
+
+    this.active = false;
+    this.queue_ = [];
+  }
+
+  init() {
+    this.actionClickHandler_ = () => this.invokeAction_();
+    this.adapter_.registerActionClickHandler(this.actionClickHandler_);
+    this.adapter_.hide();
+  }
+
+  destroy() {
+    this.adapter_.deregisterActionClickHandler(this.actionClickHandler_);
+  }
+
+  showSnackbar(data) {
+    if (data === undefined) {
+      throw new Error(
+        'Please provide a data object with at least a message to display.');
+    }
+    if (data.message === undefined) {
+      throw new Error('Please provide a message to be displayed.');
+    }
+    if (data.actionHandler && !data.actionText) {
+      throw new Error('Please provide action text with the handler.');
+    }
+
+    if (this.active) {
+      this.queue_.push(data);
+      return;
+    }
+
+    const {ACTIVE} = cssClasses;
+
+    this.adapter_.setMessageText(data.message);
+
+    if (data.actionHandler) {
+      this.adapter_.setActionText(data.actionText);
+      this.actionHandler_ = data.actionHandler;
+      this.setActionHidden_(false);
+    } else {
+      this.setActionHidden_(true);
+      this.actionHandler_ = null;
+      this.adapter_.setActionText(null);
+    }
+
+    this.active = true;
+    this.adapter_.addClass(ACTIVE);
+    this.adapter_.unhide();
+
+    setTimeout(this.cleanup_.bind(this), data.timeout || 2750);
```

Comment:
nit: could we refactor this magic number into a `numbers` getter within constants?